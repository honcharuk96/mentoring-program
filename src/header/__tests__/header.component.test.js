import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
import HeaderComponent from '../header.component';

const shallow = new ShallowRenderer();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
        pathname: 'localhost:3000/example/path',
    }),
}));[]
const openModal = jest.fn()
const setSelectedPoster = jest.fn()

describe('<HeaderComponent />', () => {

    it('should render HeaderComponent without selectedPoster', () => {
        const header =  shallow.render(<HeaderComponent openModal={openModal} setSelectedPoster={setSelectedPoster}/>);
        expect(header).toMatchSnapshot();
    });

    it('should render HeaderComponent with selectedPoster', () => {
        const header =  shallow.render(<HeaderComponent selectedPoster={1} openModal={openModal} setSelectedPoster={setSelectedPoster}/>);
        expect(header).toMatchSnapshot();
    });
});
