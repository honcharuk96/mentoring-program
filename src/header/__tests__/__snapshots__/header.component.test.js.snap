// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> should render HeaderComponent with selectedPoster 1`] = `
<React.Fragment>
  <globalStyles__Main>
    <headerstyled__TopMenu
      isPosterSelected={true}
    >
      <Memo(LogoComponent) />
      <headerstyled__SelectedImg
        alt="close"
        loading="lazy"
        onClick={[Function]}
        src="test-file-stub"
      />
    </headerstyled__TopMenu>
    <Memo(Connect(PosterInfoComponent))
      posterId={1}
    />
  </globalStyles__Main>
</React.Fragment>
`;

exports[`<HeaderComponent /> should render HeaderComponent without selectedPoster 1`] = `
<React.Fragment>
  <React.Fragment>
    <headerstyled__Header>
      <LazyImage
        alt="main"
        isBanner={true}
        src="test-file-stub"
      />
      <headerstyled__TopMenu>
        <Memo(LogoComponent) />
        <addMoviestyled__ButtonMovie
          onClick={[Function]}
        />
      </headerstyled__TopMenu>
      <Memo(Connect(Component)) />
    </headerstyled__Header>
  </React.Fragment>
</React.Fragment>
`;
